libs/portals/my-pages/health/src/screens/MedicineHistory/MedicinePrescriptionHistory.tsx

```diff
@@ -0,0 +1,212 @@
+  }>()
+
+  const { data, loading, error } = useGetMedicineHistoryQuery({
+    variables: { locale: lang },
+  })
+
+  const [
+    getMedicineDispensationsATC,
+    { loading: atcLoading, error: atcError, data: atcData },
+  ] = useGetMedicineDispensationForAtcLazyQuery({
+    fetchPolicy: 'no-cache',
+  })
+
+  const history = data?.healthDirectorateMedicineHistory.medicineHistory ?? []
+
+  useEffect(() => {
+    if (atcData) {
+      setDispensations({
+        id: atcCode,
+        data: atcData.healthDirectorateMedicineDispensationsATC.dispensations,
+      })
+    }
+  }, [atcData])
+
+  return (
+    <IntroWrapper
+      title={formatMessage(messages.medicinePrescriptionHistory)}
+      intro={formatMessage(messages.medicinePrescriptionHistoryIntroText)}
+      serviceProviderSlug={HEALTH_DIRECTORATE_SLUG}
+      serviceProviderTooltip={formatMessage(
+        messages.landlaeknirMedicinePrescriptionsTooltip,
+      )}
+      marginBottom={6}
+    >
+      {!loading && !error && history && history.length > 0 && (
+        <SortableTable
+          title=""
+          labels={{
+            medicine: formatMessage(messages.medicineTitle),
+            usedFor: formatMessage(messages.usedFor),
+            lastDispensed: formatMessage(messages.lastDispensed),
+            numberOfDispensations: formatMessage(messages.process),
+          }}
+          expandable
+          align="left"
+          defaultSortByKey="lastDispensed"
+          sortBy="descending"
+          mobileTitleKey="medicine"
+          ellipsisLength={22}
+          items={
+            history?.map((item, i) => ({
+              id: item?.id ?? `${i}`,
+              medicine: item?.name ?? '',
+              usedFor: item?.indication ?? '',
+              lastDispensed: formatDate(item?.lastDispensationDate),
+              numberOfDispensations: item.dispensationCount,
+              children: (
+                <Box padding={1} background={'blue100'}>
+                  <DispensingContainer
+                    backgroundColor="blue"
+                    label={formatMessage(messages.dispenseHistory)}
+                    showMedicineName
+                    data={(dispensations && dispensations.id === item.atcCode
+                      ? dispensations.data
+                      : item.dispensations
+                    )?.map((subItem, subIndex) => {
+                      return {
+                        pharmacy:
+                          subItem.agentName ??
+                          formatMessage(messages.notRegistered),
+                        icon: (
+                          <Icon
+                            icon={subItem?.date ? 'checkmark' : 'remove'}
+                            size="medium"
+                            color={subItem?.date ? 'mint600' : 'dark300'}
+                            type="outline"
+                          />
+                        ),
+                        date: formatDate(new Date(subItem.date)),
```

Comment:
Could the `subItem.date` ever contain an invalid date string?

Reply:
No should not, but it can be null so I added a check 🤝